pkg/googlecloud/subscriber.go

```diff
@@ -115,10 +115,12 @@ func NewSubscriber(
 ) (*Subscriber, error) {
 	config.setDefaults()
 
-	ctx, cancel := context.WithTimeout(context.Background(), config.ConnectTimeout)
-	defer cancel()
-
-	client, err := pubsub.NewClient(ctx, config.ProjectID, config.ClientOptions...)
+	clientOptions := []option.ClientOption{
+		// grpc.WithTimeout is deprecated, but pubsub.NewClient doesn't allow grpc.DialContext, only grpc.Dial
+		option.WithGRPCDialOption(grpc.WithTimeout(config.ConnectTimeout)),
+	}
+	clientOptions = append(clientOptions, config.ClientOptions...)
```

Comment:
if someone adds their own `grpc.WithTimeout` to `config.ClientOptions`, it will take precedence over `config.ConnectTimeout`.
I am not sure which should overwrite which, but there's two approaches:
1) `ClientOptions` are more specific, so if someone actually fills them in, they want those specific options. And `ConnectTimeout` defaults to 10s, so we can't rely on it overwriting `ClientOptions`
2) `ConnectTimeout` is more explicit, so it should overwrite any `ClientOptions`

I went with 1), mainly because the default value means that we don't have full control over `ConnectTimeout` and leaving an empty value would in effect overwrite an explicit command from `ClientOptions`.

I could make it so that the decision is made before `config.SetDefaults()`, I don't know.
What are your opinions?

Reply:
Actually, the problem is in that there's two ways to do the exact same thing. But I think that setting the `ClientOption` by hand might be too obscure for an end user, so it might be useful to expose it directly from the config (?)